scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -581,8 +581,9 @@ export class Editor3Component extends React.Component<IPropsEditor3Component, IS
          * add much performance overhead if it was replaced unconditionally, but I don't want to break it
          * nor spend time on testing so I'm keeping it as is for now.
          */
-        const dropAreaEnabled =
-            this.props.editorFormat.includes('media') || this.props.editorFormat.includes('embed articles');
+        const dropAreaEnabled = this.props.editorFormat.some(
+            (option) => formattingOptionsThatRequireDragAndDrop.has(option),
+        );
```

Comment:
Might be good to extract this as a boolean at this point and reuse in `scripts/core/editor3/components/BaseUnstyledComponent.tsx:54`

Reply:
It requires an argument so it can't be a boolean. It could be a function, but other usage is a bit different, so I think having a constant for formatting options is fine.